src/util.js

```diff
@@ -118,4 +120,47 @@ function letMixin(Type) {
   return Type;
 }
 
-export { deprecate, serviceResponse, letMixin };
+/**
+ * Utility function for determining if detect-browser package successfully set
+ * itself up and determined the platform
+ */
+function canDeterminePlatform() {
+  return !!platform;
+}
+
+/**
+ * Utility function to get the name of the platform running the code. For a
+ * browser, it will be the browser's name; otherwise, it will be 'node'.
+ */
+function getPlatformName() {
+  return !!platform && platform.name;
+}
+
+/**
+ * Utility function to get the version of the platform running the code.
+ */
+function getPlatformVersion() {
+  return !!platform && platform.version;
+}
+
+/**
+ * Utility function to encode a JSON object into a valid string that can be
+ * inserted in a URI. The object is first stringified, then encoded in base64,
+ * and finally encoded via the builtin encodeURIComponent function.
+ *
+ * @param {Object} obj The object to encode
+ * @returns {String} The encoded object
+ */
+function uriEncodeObject(obj) {
+  return encodeURIComponent(new Buffer(JSON.stringify(obj)).toString('base64'));
```

Comment:
Sadly `Buffer` won't work in the browser. We may need a base64 encode/decode polyfill to do this in a portable way, such as https://github.com/davidchambers/Base64.js/.